src/reducers/restaurants.js

```diff
@@ -0,0 +1,15 @@
+const initialState = []
+
+
+const restaurants = (state = initialState, action) => {
+    switch (action.type) {
+
+        case 'GET_RESTAURANTS':
+            return state = action.payload
```

Comment:
Muito cuidado, o que está sendo retornado aqui é uma atribuição `state = action.payload`, uma atribuição sempre é `undefined`. Então sempre que o reducer for chamado com a ação do tipo `GET_RESTAURANT` ele vai apagar todo o estado e colocar `undefined` no lugar.